src/components/Header/Header.test.js

```diff
@@ -1,22 +1,24 @@
 import React from 'react';
 import { render, screen } from '@testing-library/react';
+import { BrowserRouter as Router } from 'react-router-dom'
 import Header from './Header';
+import { ContextWrapper } from '../../states/GlobalContext';
 
 describe('Header works properly', () => {
   it('renders the menu and login buttons', () => {
-    render(<Header />);
+    render(<ContextWrapper><Router><Header /></Router></ContextWrapper>);
```

Comment:
When you need to nest the component you want to test into providers, there's a way of doing for all the tests. This way you won't need to do this for each test case. Check the following documentation about custom rendering and let me know if you have any questions:

https://testing-library.com/docs/react-testing-library/setup/#custom-render